api/app/controllers/spree/api/v2/platform/payment_methods_controller.rb

```diff
@@ -0,0 +1,29 @@
+module Spree
+  module Api
+    module V2
+      module Platform
+        class PaymentMethodsController < ResourceController
+          private
+
+          def model_class
+            Spree::PaymentMethod
+          end
+
+          def spree_permitted_attributes
+            preferred_attributes = []
+
+            if action_name == 'update'
```

Comment:
@MatthewKennedy why only update? we should also allow setting preferences on create I suppose

Reply:
@damianlegawiec Until the payment method has been created and given a payment method type it doesn't have any `.defined_preferences`

Reply:
I'll move this up to the resource controller and write it to use `return unless defined_preferences.present? ` that way if other resources don't use STI and do have preferences by default they will work on create.

Reply:
@MatthewKennedy these preferences are becoming more of an issue every day, I suppose we need to migrate it into JSONb ASAP (next release), let's chat about it today maybe?

Reply:
@damianlegawiec I agree, moving the the rails active record store on a JSONb column needs doing. Yeah we can discuss this today.